### ISSUE TITLE
about merge most common classes

### ISSUE THREAD

**Issue by xiaoyuhao1995** [Nikronic/ObjectNet#4]
hi,Nikronic!

I have a question for you about ObjectNet. How did you merge most common classes to reduce number of classes from 150 to 25?

**Comment by Nikronic** [comment]
Hi @xiaoyuhao1995 
Actually, I am not using any statistical or learning method to achieve this reduction. 
My goal was to use this model as a module in an image refinement task object-wisely. But I did not want to train or do any time/memory consuming operation on this network. So, I considered myself as an expert(!) and combined classes similar to each other based on common sense.

Here is the explanation:
1. The original output of the model is the output of softmax function, so summation of all the probabilities for 150 classes will be 1. 
2. There are some classes which are similar. For instance, row 30 and 10 in [classes info file](https://github.com/Nikronic/ObjectNet/blob/master/data/object150_info.csv) and to achieve a combined probability for these combined classes, I just added their probabilities and define a new class containing all similar classes.

It seems I'd forgotten to add the combination info. See commit https://github.com/Nikronic/ObjectNet/commit/c8fd6034474ac662cfe9bce8cabf3350b79ffb17 please for new files regarding this info.

[combination file](https://github.com/Nikronic/ObjectNet/blob/master/data/object25_info.csv) contains the combined classes' row IDs and [mapping function](https://github.com/Nikronic/ObjectNet/blob/master/data/mapped_25info.txt) contains the mapping of file1's combinations into new indices.

**Comment by xiaoyuhao1995** [comment]
Thanks @Nikronic 
So you just use ObjectNet in eval mode to get the 25 class feature map.And in the process of learning the weights of ObjectNet was not update. Is that right?

**Comment by Nikronic** [comment]
@xiaoyuhao1995 yes exactly.

**Comment by xiaoyuhao1995** [comment]
@Nikronic hava you try to combine the Coarsenet,Objectnet,Edgenet and Detailset to train? recently I try to combine those network and I find it is difficult. Maybe we can talk about it.

**Comment by Nikronic** [comment]
@xiaoyuhao1995 Not really. First of all, CoarseNet is not working properly, literally taking an average filter over image would do same thing as CoarseNet is doing. Secondly, DetailsNet has not been finished as in term of mathematically, I am not sure how two discriminator is going to act properly, and because of that, it's generating completely nonsense outputs. Finally, any combination of those networks won't lead to anything meaningful unless the aforementioned problem has been solved.

In the end, currently I am focusing on my master's thesis so I have no time for revising this project. Maybe in Summer.
